### ISSUE TITLE
Accidental screen switches

### ISSUE THREAD

**Issue by mkellyxp** [InfiniTimeOrg/InfiniTime#618]
I've noticed this happen when wake is on wrist raise, or tap.  I'll go to look at my watch and it's in the apps or settings screen.  My guess is that something touches the watch and it switches screens without me knowing.  

Could there be a way to lock it on that home screen unless you double tap to unlock or something.  Not sure if this would be useful to others, but I know 99.9% of my use of this is just looking at that home screen

**Comment by adocampo** [comment]
Same here, is a bit annoying wanting to see the time and finding yourself pushing the back button several times until reach the home screen, because the watch is inside settings or main menu, a couple of times, even it woke at the pong and the paint application which I never opened.
I'm on 1.4 and have both wake on wrist and 2 tap wake (tried also with 1-tap wake and the same issue is present)

**Comment by RageGamerBoi** [comment]
I’m gonna put a part of my suggestion from #1390 here:

If a navigation screen is opened and the watch goes to sleep because it isn’t being used, it should return to the watch face.
This shouldn’t apply inside of apps, or for a few seconds immediately after the watch goes to sleep, because the page might need to be used again.

(read the complete thing in that issue, it goes into a little bit more detail, but I think you get the idea)

**Comment by bradsk88** [comment]
There was some concern that "we don't know what the user wants to see on wake".  But that might be remediable by making this an option:
```
On wake:
- Show last open app
- Return to home
```

**Comment by Riksu9000** [comment]
I don't think anyone would prefer for it to always return home. #1592 covers all intended use cases where you can't see the time. This issue is about screen switches that should not happen in the first place.

**Comment by RageGamerBoi** [comment]
That’s why I think there is some elegance to the “switch to watch face from only menu screens and a few seconds after sleep”…
Because they’re menus it’s only a few swipes and it wouldn’t interrupt anything important, while also fixing the majority of instances in which this problem occurs.
The only place this doesn’t currently apply is the Settings menu, and that can simply be omitted from the category of “menu screens” for this purpose.
This also (in some ways) can address concerns about readability of the small pop up time from #1592

**Comment by etewassad** [comment]
I don't agree with the return to home screen solution either but can we add some kind of lock function for raise to wake users? I find the watch ending up in the setting screen randomly a lot.

Single tap or double tap something like that, I understand it needs to be unintrusive. I'd like to hear more people's input.

**Comment by RageGamerBoi** [comment]
> Another nice (configurable) option would be if we could set the screen to "wake" on raise, but require a physical button press to interact.

I believe this is the best solution, as accidental touches are another thing that happens frequently, so simply having a "tap to unlock" feature wouldn't be as effective as intended. It should obviously be disabled in apps such as the stopwatch and timer.

Returning to the home screen from menu screens is a good thing to include alongside this, as it is very possible for the watch to be left on a menu screen simply because the user forgot to swipe back. It could be toggle-able as well if there is a use-case in which this would not be beneficial (though I can't personally think of any).

**Comment by PlaidPhantom** [comment]
I'd love something like this to prevent accidental presses; just this morning I somehow accidentally changed my watch face. What about an optionally-enabled "lock" screen for raise/shake wakeups that just shows the current watch face (plus a lock icon or something)? Then, its only available action would be to resume to whatever the current wakeup behavior is on button press. With that, changing the wakeup screen behavior could be a separate discussion.
